### PR TITLE
Export System & Dataset as csv

### LINKED ISSUE #299: Export existing fidesctl resource metadata to csv file
In an organization, it would be helpful to export the rich metadata that fidesctl already captures in Registry, System, and Dataset resources so that it can provide the foundation for a data map (otherwise known as a data inventory, or data flow map). 

This issue will solve for the target state experience:
* fidesctl developer user runs fidesctl export command, ie `fidesctl export path/to/folder/`
* .csv file is written to specified folder that contains Organization, System, Registry, and Dataset metadata, which includes:
  * organization.name, address, email, phone
  * organization.dponame, dpoaddress, dpoemail, dpophone
  * organization.representative, repaddress, repemail, repphone
  * organization.securitypolicy (url)
  * system, dataset: export only what we already capture (resource name, use, subject type, data categories, qualifier)

This will be complete when the CSV file looks similar to the csv here:  https://docs.google.com/spreadsheets/d/1AItjzt2DOvCyG9my2wDlUkiTFX8wr9w1gLpNG8VzwTE/edit#gid=0

### PR DESCRIPTION
**Pull request by SteveDMurphy**
Closes #299 

### Code Changes

* [x] Allow grouped commands (i.e. `fidesctl export system`)
* [x] Add an `export` command to facilitate exporting a data map and ropa (eventually)

### Steps to Confirm

* [x] csv exports look as they should per the examples provided

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated

### Description Of Changes

As part of the crawl state, we want to be able to at least export some of the existing resources currently maintained in fides as csv.

I spent some time (probably too much tbf) looking at different options to try and flatten out the nested lists of data into something that aligned with the examples provided in the design/feature docs. Initially I thought this would be fairly straight forward to do using `json_normalize` in [pandas](https://pandas.pydata.org/docs/reference/api/pandas.json_normalize.html#). It ended up looking a bit over complicated by the time I had it (mostly) working so ended up using a list of tuples as records. I am not certain this is the right path forward by any means but as a starting point it felt the most clear. I'd love to learn more about if there is a way to use the default taxonomy class better or really what the rest of the team thinks in general

### CODE REVIEW

**Comment by SteveDMurphy** [comment]
I think this is ready for an early review if anyone has any time @ethyca/fides-control - I still need to add some documentation around the commands but hesitated at the last since it sounds like these initial commands may be somewhat temporary...

These commands should work with the existing resources defined in `fides_resources` and demo_resources`. 

1) Apply the resources if necessary as the export data is taken from the server
`fidesctl apply demo_resources` 
2) Export the system and/or dataset, I ended up using the `manifest_dir` as the output location for now, I can imagine a world where we have a standard location for output based on some of the `fidesctl init` work in #313 
`fidesctl export system demo_resources`
`fidesctl export dataset demo_resources`

3) Additionally, a dry flag was added that prints the csv records instead of creating a file (primarily for testing the cli)
`fidesctl export dataset demo_resources --dry`

Added some notes in the PR description, but essentially I was looking at using pandas to help with some of the export complexity but it felt too convoluted and difficult to iterate with over the coming issues. Based on that, I ended up looking at using tuples as records and defining the output headers as a tuple itself. If there are any thoughts on how to do this differently, something I missed completely somewhere, or suggestions of any sort would love to hear them - thank you!

File: fidesctl/src/fidesctl/cli/__init__.py [hunk 1]
```diff
@@ -40,7 +42,6 @@
 @click.group(
     context_settings=CONTEXT_SETTINGS,
     invoke_without_command=True,
-    chain=True,
```
Comment (ThomasLaPiana) [inline fidesctl/src/fidesctl/cli/__init__.py:43]: i did this on purpose, and i think it was to make the auto-generated docs work. can you confirm this doesn't break them?

File: fidesctl/src/fidesctl/core/export.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+from datetime import datetime
+from typing import Dict, List, Tuple, Set
+
+import csv
+
+from fidesctl.core.api_helpers import get_server_resources
+from fidesctl.core.utils import echo_green, get_all_level_fields
+
+
+def list_fides_keys(resources: List) -> List:
```
Comment (ThomasLaPiana) [inline fidesctl/src/fidesctl/core/export.py:11]: this seems like too small of a thing to be its own function, was there a case that necessitated it?

File: fidesctl/src/fidesctl/core/export.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    """
+    Returns a list of fides_keys found in the manifest dir
```
Comment (ThomasLaPiana) [inline fidesctl/src/fidesctl/core/export.py:13]: `found in the resource list`

File: fidesctl/src/fidesctl/core/export.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    """
+    Returns a list of tuples as records (with expected headers)
+    to be exported as csv
+    """
+
+    output_list = [
+        (
+            "dataset.name",
+            "dataset.description",
+            "dataset.data_categories",  # from Kelly - unique data categories per data qualifier, per data use per for the entire dataset, per row.
```
Comment (ThomasLaPiana) [inline fidesctl/src/fidesctl/core/export.py:74]: can you move this context to the docstring?

**Comment by ThomasLaPiana** [comment]
@SteveDMurphy from reading through your code, i totally support the tuple method and think it looks great, I had no problem following along with what was going on and how things were being constructed. Pandas is a very heavy requirement, so using it just to dump out some CSVs seems inefficient if we can do without it.

After a quick review this PR looks great, once the last tweaks are in I'm happy to do a full review on it

File: fidesctl/src/fidesctl/cli/__init__.py (hunk 1 above)
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/cli/__init__.py:43]: I had heard that somewhere recently and ended up testing it (was looking at this as part of the initial `scan` command) - tested using `make docs-serve` to generate the docs locally, is that the best way to confirm @ThomasLaPiana ?

File: fidesctl/src/fidesctl/core/export.py (hunk 2 above)
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/core/export.py:11]: Mostly just around the DRY aspect when planning for separate commands, I think I originally saw we were doing something similar in a couple of other places as well so considered shifting this into `fidesctl.core.utils` for a future refactor and consistency. Happy to get rid of it and build the list separately in each, definitely see what you mean about being to small to warrant its own function 👍🏽

File: fidesctl/src/fidesctl/core/export.py (hunk 4 above)
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/core/export.py]: Fixed! Thank you 🙌🏽

File: fidesctl/src/fidesctl/core/export.py (hunk 3 above)
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/core/export.py]: deleted the function but great catch, thanks for clarifying that for my brain!

**Comment by SteveDMurphy** [comment]
> @SteveDMurphy from reading through your code, i totally support the tuple method and think it looks great, I had no problem following along with what was going on and how things were being constructed. Pandas is a very heavy requirement, so using it just to dump out some CSVs seems inefficient if we can do without it.
> 
> After a quick review this PR looks great, once the last tweaks are in I'm happy to do a full review on it

Thanks for the early feedback @ThomasLaPiana ! Changes made and moving to ready for review, could use an extra set of eyes that what I check for auto-generating the docs was correct however. 

Also, I noticed that pandas is already listed in [requirements.txt](https://github.com/ethyca/fides/blob/e36b2c24f9de3aec2622ef44c2a915000b8aa89c/fidesctl/requirements.txt#L7) but I couldn't see any obvious usage - ok to remove?

**Comment by SteveDMurphy** [comment]
> Also, I noticed that pandas is already listed in [requirements.txt](https://github.com/ethyca/fides/blob/e36b2c24f9de3aec2622ef44c2a915000b8aa89c/fidesctl/requirements.txt#L7) but I couldn't see any obvious usage - ok to remove?

Discussed in meeting, required for plotly usage as of today 👍🏽

File: fidesctl/src/fidesctl/cli/__init__.py (hunk 1 above)
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/cli/__init__.py:43]: Per earlier conversation @ThomasLaPiana - I _think_ the auto-generated docs are still working looking at this:
image

Comment (ThomasLaPiana) [reply fidesctl/src/fidesctl/cli/__init__.py:43]: sounds good, i'll take a look and double check but thats good news!

**Comment by ThomasLaPiana** [comment]
@SteveDMurphy i made some small changes, lmk what you think! overall looks excellent.

Did you want to leave documentation changes to the walk/run state? This seems a little thin to document, especially considering it'll change a lot

**Review by ThomasLaPiana — APPROVED** [review]
LGTM!

**Comment by SteveDMurphy** [comment]
> @SteveDMurphy i made some small changes, lmk what you think! overall looks excellent.
> 
> Did you want to leave documentation changes to the walk/run state? This seems a little thin to document, especially considering it'll change a lot

Thanks for the updates everywhere @ThomasLaPiana ! All look good to me 🙌🏽 

As far as the docs, I was thinking along the same lines with the expected churn and sort of minimum functionality so far 👍🏽

**Pull request merged**